go/pkg/apis/enricher/framework/python/flask_detector.go

```diff
@@ -44,6 +46,89 @@ func (d FlaskDetector) DoFrameworkDetection(language *model.Language, files *[]s
 	}
 }
 
+// getPortFromFileFlask tries to find a port configuration inside a given file content
+func getPortFromFileFlask(matchIndexRegexes []model.PortMatchRule, text string) []int {
+	var ports []int
+	for _, matchIndexRegex := range matchIndexRegexes {
+		matchIndexesSlice := matchIndexRegex.Regex.FindAllStringSubmatchIndex(text, -1)
+		for _, matchIndexes := range matchIndexesSlice {
+			if len(matchIndexes) > 1 {
+				port := getPortWithMatchIndexesFlask(text, matchIndexes, matchIndexRegex.ToReplace)
+				if port != -1 {
+					ports = append(ports, port)
+				}
+			}
+		}
+	}
+
+	return ports
+}
+
+func getPortWithMatchIndexesFlask(content string, matchIndexes []int, toBeReplaced string) int {
+	// select the correct range for placeholder and remove unnecessary strings
+	portPlaceholder := content[matchIndexes[0]:matchIndexes[1]]
+	portPlaceholder = strings.Replace(portPlaceholder, toBeReplaced, "", -1)
+	// try first to check for hardcoded ports inside the app.run command. e.g port=3001
+	re, err := regexp.Compile(`port=*(\d+)`)
+	if err != nil {
+		return -1
+	}
+	if port := utils.FindPortSubmatch(re, portPlaceholder, 1); port != -1 {
+		return port
+	}
+
+	// get the value of kwarg "port" inside app.run and check if there is
+	// any variable defined with its value
+	contentBeforeMatch := content[0:matchIndexes[0]]
+	portPlaceholder = strings.Replace(portPlaceholder, "port=", "", -1)
+	re, err = regexp.Compile(portPlaceholder + `\s=\s*(\d+)`)
+	if err != nil {
+		return -1
+	}
+	matches := re.FindStringSubmatch(contentBeforeMatch)
+	if len(matches) > 0 {
+		portValue := matches[len(matches)-1]
+		re, err = regexp.Compile(`:*(\d+)$`)
+		if err != nil {
+			return -1
+		}
+		if port := utils.FindPortSubmatch(re, portValue, 1); port != -1 {
+			return port
+		}
+	}
+
+	return -1
+}
+
+// DoPortsDetection searches for the port in app/app.py or /wsgi.py
```

Comment:
it looks like we only search for app.py in the root. do we also look in app/ directory?